htlcswitch/switch.go

```diff
@@ -781,6 +795,10 @@ func (s *Switch) handleLocalDispatch(pkt *htlcPacket) error {
 		payment.preimage <- zeroPreimage
 		s.removePendingPayment(pkt.incomingHTLCID)
 
+		if err := s.control.Fail(pkt.circuit.PaymentHash); err != nil {
```

Comment:
Not sure if we should worry about this failing. Let's say the payment and its status update fail. If someone attempts to send this payment again, they won't be able to because the payment is still in flight.

Reply:
this one was described in original ticket

Reply:
This may need to be moved into `parseFailPayment`. This would allow us to discern errors that are sourced locally vs remote. If it's a locally sourced error, then we don't need to update the state as the HTLC never left the daemon.